### PR TITLE
feat:remove jprotoc dependency

### PR DESCRIPTION
**Pull request by DreamerPXY**
## What is the purpose of the change
dubbo-compiler remove jprotoc dependecy

## Brief changelog
1. Remove jprotc dependency and add mustache dependency for template generation code
2. Retain the ProtoTypeMap provided by the original jprotc for storing message types parsed from proto
3. Remove ProtocPlugin and use Dubbo's own DubboProtocPlugin as the main entry point for protoc plugin calls
4. DubboProtocPlugin no longer provides debug methods and only supports FEATURE_ PROTO3_ OPTIONAL

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-compiler/pom.xml [hunk 1]
```diff
@@ -37,8 +37,9 @@
 
     <dependencies>
         <dependency>
-            <groupId>com.salesforce.servicelibs</groupId>
-            <artifactId>jprotoc</artifactId>
+            <groupId>com.github.spullara.mustache.java</groupId>
```
Comment (EarthChen) [inline dubbo-compiler/pom.xml:40]: format

File: dubbo-compiler/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+## dubbo-complier移除jpotoc依赖
```
Comment (EarthChen) [inline dubbo-compiler/README.md:1]: pls use english

**Review by EarthChen — CHANGES_REQUESTED** [review]

**Comment by AlbumenJ** [comment]
@guohao PTAL

File: dubbo-compiler/src/main/java/org/apache/dubbo/gen/AbstractGenerator.java [hunk 3]
```diff
@@ -261,6 +261,20 @@ private List<PluginProtos.CodeGeneratorResponse.File> buildFile(ServiceContext c
...
+        if (resource == null) {
+            throw new RuntimeException("Could not find resource " + resourcePath);
+        } else {
+            InputStreamReader resourceReader = new InputStreamReader(resource, Charsets.UTF_8);
+            Mustache template = MUSTACHE_FACTORY.compile(resourceReader, resourcePath);
+            return template.execute(new StringWriter(), generatorContext).toString();
+        }
+    }
+
+
```
Comment (dongxuwang) [inline dubbo-compiler/src/main/java/org/apache/dubbo/gen/AbstractGenerator.java:277]: I think this is the redundant blank line needs to be removed

**Comment by EarthChen** [comment]
relate https://github.com/apache/dubbo/issues/12381

**Pull request closed without merging**